app/code/core/Mage/SalesRule/Model/Validator.php

```diff
@@ -962,7 +962,7 @@ protected function _addDiscountDescription($address, $rule)
         $label = '';
         if ($ruleLabel) {
             $label = $ruleLabel;
-        } elseif (strlen($address->getCouponCode())) {
+        } elseif (strlen($address->getCouponCode() ?? '')) {
```

Comment:
`coupon_code` getter in Quote model was overridden in #2825 to always return empty string, maybe a similar approach can be done with address.

Reply:
@elidrissidev I think I did it the right way now